patches/tModLoader/Terraria/ModLoader/TileLoader.cs

```diff
@@ -1016,4 +1018,80 @@ public static bool UnlockChest(int i, int j, int type, ref short frameXAdjustmen
 	{
 		return GetTile(type)?.UnlockChest(i, j, ref frameXAdjustment, ref dustType, ref manual) ?? false;
 	}
+
+	public static void RecountTiles(SceneMetrics metrics)
+	{
+		// reset every tile count
+		metrics.HolyTileCount = metrics.EvilTileCount = metrics.BloodTileCount = metrics.SnowTileCount = metrics.JungleTileCount = metrics.MushroomTileCount = metrics.SandTileCount = 0;
+
+		// loop through all tiles, skipping ones not onscreen, and add each to the biome tile counts from their respective sets
+		for (int i = 0; i < TileCount; i++) {
+
+			int tileCount = metrics._tileCounts[i];
+
+			if (tileCount == 0)
+				continue;
+
+			int hallow = TileID.Sets.HallowBiome[i];
+			if (hallow != 0)
+				metrics.HolyTileCount += tileCount * hallow;
+
+			int snow = TileID.Sets.SnowBiome[i];
+			if (snow != 0)
+				metrics.SnowTileCount += tileCount * snow;
+
+			int mushroom = TileID.Sets.MushroomBiome[i];
+			if (mushroom != 0)
+				metrics.MushroomTileCount += tileCount * mushroom;
+
+			int sand = TileID.Sets.SandBiome[i];
+			if (sand != 0)
+				metrics.SandTileCount += tileCount * sand;
+
+			int crimson, corrupt, jungle = 0;
+
+			// handles if the world is using the remix seed or not, which slightly changes which blocks count
+			if (!Main.remixWorld) {
+				corrupt = TileID.Sets.CorruptBiome[i];
+				crimson = TileID.Sets.CrimsonBiome[i];
+				jungle = TileID.Sets.JungleBiome[i];
+			}
+
+			else {
+				corrupt = TileID.Sets.RemixCorrupt[i];
+				crimson = TileID.Sets.RemixCrimson[i];
+				jungle = TileID.Sets.RemixJungle[i];
+			}
+
+			if (corrupt != 0)
+				metrics.EvilTileCount += tileCount * corrupt;
```

Comment:
No need for the `if`. The `+=` and `*` are quite fast, and won't do anything if the value is 0 anyway